detect_secrets/core/usage.py

```diff
@@ -44,6 +74,36 @@ def add_no_verify_flag(parser):
     )
 
 
+def add_shared_arguments(parser):
+    """
+    These are arguments that are for both
+    pre-commit and regular detect-secrets scripts.
+    """
+    add_exclude_lines_argument(parser)
+    add_word_list_argument(parser)
+    add_custom_plugins_argument(parser)
+    add_use_all_plugins_argument(parser)
+    add_no_verify_flag(parser)
+
+
+def get_parser_to_add_opt_out_options_to(parser):
+    """
+    The pre-commit hook gets e.g. `--no-jwt-scan` type options
+    as well as the subparser for `detect-secrets scan`.
+
+    :rtype: argparse.ArgumentParser
+    :returns: argparse.ArgumentParser to pass into PluginOptions
+    """
+    if parser.prog == 'detect-secrets-hook':
+        return parser
+
+    for action in parser._actions:  # pragma: no cover (Always returns)
+        if isinstance(action, argparse._SubParsersAction):
+            for subparser in action.choices.values():
+                if subparser.prog.endswith('detect-secrets scan'):
```

Comment:
Is this for testing purpose? The main reason I ask is because during plugin dev phase, I would use something like below to test out the code without installing it.

```shell
env PYTHONPATH=/Users/username/workspace/detect-secrets python /Users/username/workspace/detect-secrets/detect_secrets/main.py scan tmp/testfile.txt
```

The hard code part here ruined the execution of my case.

Reply:
I wonder if I change it to just `endwith('scan')`, if it'll work, it probably does.

Reply:
You are right, changing it `endwith('scan')` is working! @KevinHock